tc/tc_util.c

```diff
@@ -205,13 +206,20 @@ int parse_percent_rate(char *rate, const char *str, char *dev)
 	if (read_prop(dev, "speed", &max_rate_bits))
 		return -1;
 
-	ret = sscanf(str, "%lf%%", &perc);
-	if (ret != 1) {
-		fprintf(stderr, "Specified rate value could not be read or is malformed\n");
-		return -1;
-	}
+	ret = sscanf(str, "%m[0-9.%]", &str_perc);
+	if (ret != 1)
+		goto malf;
+
+	/* Make sure there's only one percent sign and it's at the end */
+	perc = strtod(str_perc, &p);
+	if (*p != '%' || *(p++) != '\0')
+		goto malf;
 
-	if (perc > 100.0 || perc < 0.0) {
+	saved_errno = errno;
+	free(str_perc);
+
+
```

Comment:
Just remove one of these extra blank lines and I think this patch will be good to go.